### PR TITLE
add HMap.mapValues

### LINKED ISSUE #39: add HMap.mapValues
it should take a FunctionK

### PR DESCRIPTION
**Pull request by alexeygorobets**
close #39

### CODE REVIEW

File: core/src/main/scala/com/stripe/dagon/HMap.scala [hunk 1]
```diff
@@ -81,6 +81,9 @@ final class HMap[K[_], V[_]](protected val map: Map[K[_], V[_]]) extends Seriali
     val fnAny = f.toFunction[Any].andThen(_.iterator)
     map.iterator.asInstanceOf[Iterator[(K[Any], V[Any])]].flatMap(fnAny).toStream
   }
+
+  def mapValues[V1[_]](f: FunctionK[V, V1]): HMap[K, V1] =
+    HMap.from[K, V1](map.mapValues(f(_)))
```
Comment (johnynek) [inline core/src/main/scala/com/stripe/dagon/HMap.scala]: Actually one request, mapValues is lazy on maps and this often burns people. Can we make this strict? `map.iterator.map { case (k, v....` or something?

Comment (alexeygorobets) [reply core/src/main/scala/com/stripe/dagon/HMap.scala]: sure!

**Comment by johnynek** [comment]
Thanks!

**Comment by johnynek** [comment]
Oh, it looks like git can’t tell this is independent of your previous PR and there is a merge conflict. :(

**Pull request merged**